IDT.MVNO/Main/UI/Retailers/Components/UIElementBasket.cs

```
@@ -0,0 +1,42 @@
+﻿namespace IDT.MVNO.Main.UI.Retailers.Components
+{
+    using IDT.Core.Automation.Page.Base;
+    using System.Collections.Generic;
+    using OpenQA.Selenium;
+    using IDT.Core.Automation.Utilities.CommonAction;
+
+    /// <summary>
+    /// This class will contain all webelements from menus
+    /// </summary>
+    public class UIElementBasket : BasePage
+    {
+        private IDictionary<string, IWebElement> MapTextWebElement;
+
+        public UIElementBasket()
+        {
+            this.MapTextWebElement = new Dictionary<string, IWebElement>();
+        }
+
+        public void AddElement(string elementName, IWebElement element)
+        {
+            if (!this.MapTextWebElement.ContainsKey(elementName))
+            {
+                this.MapTextWebElement.Add(elementName, element);
+            }
+        }
+
+        public IWebElement GetWebElement(string elementName)
+        {
+            return this.MapTextWebElement[elementName];
+        }
+
+        public void ClickOnOption(string elementText)
```

I think that we should raise an exception is element is not found here